### ISSUE TITLE
AuthUser Doesn't work

### ISSUE THREAD

**Issue by cfristofer** [sochix/TLSharp#48]
it connect ok, but when a trying to make AuthUser it gave this error  "Your phone number registered to 1 dc. Please update settings. See https://github.com/sochix/TLSharp#i-get-an-error-migrate_x for details.". I use all these, but still doesn't work 
i checked these server:
Server 1: 149.154.175.50:443
Server 2: 149.154.167.51:443
Server 3: 149.154.175.100:443
Server 4: 149.154.167.91:443
Server 5: 91.108.56.165:443
server 6:149.154.167.40:443
server 7:149.154.167.50:443

and set phone numbers in app.configs but doesn't work,
how can i create Session.dat file?

**Comment by cfristofer** [comment]
I solve it, theres any way to get new messages from any chat ??

**Comment by dinhducit** [comment]
can you run AuthUser(), I got exception at this function, can you solve it?

**Comment by cfristofer** [comment]
yeah, what is the exception you got ?? the same i put before ??

**Comment by dinhducit** [comment]
I get an exception: System.IO.EndOfStreamException: Unable to read beyond the end of the stream. :(

**Comment by cfristofer** [comment]
I get an exception: System.IO.EndOfStreamException: Unable to read beyond the end of the stream. All test methos except that AuthenticationWorks and TestConnection return same error. I did every thing including setting api id and hash, and setting server address.

You should create a Telegram session. See configuration guide

**Comment by cfristofer** [comment]
I didn't get that error, i don't know how to solve it. But https://github.com/sochix/TLSharp here says how to

**Comment by dinhducit** [comment]
so,  can you send message as well?

**Comment by siml173** [comment]
my phone number is registered in telegram with android app. when im trying to makeauth it return following error : 
PHONE_NUMBER_UNOCCUPIED
please help

**Comment by cfristofer** [comment]
Yeah, i can send a message as well, you're trying to implement this at your work or at home ??

**Comment by cfristofer** [comment]
I did not get that error before, when i have a chance i'll investigate for a while to see if  i can help you .

**Comment by siml173** [comment]
im working at home, telegram sent code to my number successfully, but in makeauth method i got that error.
thank you for your attention, :-)

**Comment by sochix** [comment]
See telegram api reference here https://core.telegram.org/method/auth.signIn
tldr; 
400 BAD_REQUEST	PHONE_NUMBER_UNOCCUPIED	The code is valid but no user with the given number is registered

**Comment by siml173** [comment]
my phone number is registered in telegram with android app.

**Comment by sochix** [comment]
Try another DC

**Comment by siml173** [comment]
what is "DC" ?  . im sorry for my questions. and thank you so mush for this great source code. i learned many thing . thank you again,

**Comment by sochix** [comment]
Data center. See here https://github.com/sochix/TLSharp#i-get-an-error-migrate_x

**Comment by siml173** [comment]
hi.
in following method : OnResponse in class : AuthSendCodeRequest
in following lines : 
_phoneRegistered = reader.ReadUInt32() == boolTrue;
_phoneCodeHash = Serializers.String.read(reader);
it return an exception with following content : 
unable to read beyond the end of stream.
please help.

**Comment by siml173** [comment]
in register form in telegram api, i get Public keys,  should i put it some where?

**Comment by siml173** [comment]
please help. im realy stucked on this
